metric/builds.go

```diff
@@ -59,57 +73,68 @@ func FailedBuildCount() *prometheus.CounterVec {
 			Name: "harness_ci_pipeline_execution_errors_total",
 			Help: "Total number of pipeline executions which failed due to system errors",
 		},
-		[]string{"pool_id", "os", "arch", "driver"},
+		[]string{"pool_id", "os", "arch", "driver", "distributed"},
 	)
 }
 
 // RunningCount provides metrics for number of builds currently running
-func RunningCount(instanceStore store.InstanceStore) *prometheus.GaugeVec {
+func RunningCount() *prometheus.GaugeVec {
 	return prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
 			Name: "harness_ci_pipeline_running_executions",
 			Help: "Total number of running executions",
 		},
-		[]string{"pool_id", "os", "arch", "driver", "state"}, // state can be running, in_use, or hibernating
+		[]string{"pool_id", "os", "arch", "driver", "state", "distributed"}, // state can be running, in_use, or hibernating
 	)
 }
 
 // RunningPerAccountCount provides metrics at account level for running executions
 // This might be removed in the future as we don't want labels with high cardinality
 // We are just using two labels at the moment which should be pretty small
-func RunningPerAccountCount(instanceStore store.InstanceStore) *prometheus.GaugeVec {
+func RunningPerAccountCount() *prometheus.GaugeVec {
 	return prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
 			Name: "harness_ci_pipeline_per_account_running_executions",
 			Help: "Total number of running executions per account",
 		},
-		[]string{"owner_id", "os"},
+		[]string{"owner_id", "os", "distributed"},
 	)
 }
 
-func updateRunningCount(ctx context.Context, instanceStore store.InstanceStore,
-	runningMetric *prometheus.GaugeVec, runningPerAccountMetric *prometheus.GaugeVec) {
-	for {
-		time.Sleep(dbInterval)
-		m := make(map[label]int)
-		// collect stats here
-		instances, err := instanceStore.List(ctx, "", &types.QueryParams{})
-		if err != nil {
-			// TODO: log error
-			continue
-		}
-		runningMetric.Reset()
-		runningPerAccountMetric.Reset()
-		for _, i := range instances {
-			l := label{os: i.OS, arch: i.Arch, state: string(i.State), poolID: i.Pool, driver: string(i.Provider)}
-			if i.OwnerID != "" {
-				runningPerAccountMetric.WithLabelValues(i.OwnerID, i.OS).Inc()
+func (m *Metrics) UpdateRunningCount(ctx context.Context) {
+	go func() {
+		for {
+			time.Sleep(dbInterval)
+			wg := &sync.WaitGroup{}
+			for _, ms := range m.stores {
```

Comment:
why do we need multiple stores here and a wait group?
For single mode, it would iterate over sqlite and for distributed mode over postgres - only difference would be the way it's querying the running builds right?